src/core/utils/getTokenMetadata.ts

```diff
@@ -71,22 +66,30 @@ async function getOnChainTokenMetadata(
   const tokenContract = Erc20__factory.connect(tokenAddress, provider)
 
   try {
-    const name = await fetchStringTokenData(tokenContract, provider, 'name')
-    const symbol = await fetchStringTokenData(tokenContract, provider, 'symbol')
-    const decimals = Number(await tokenContract.decimals())
+    const [name, symbol, decimal] = await Promise.all([
+      fetchStringTokenData(tokenContract, provider, 'name'),
+      fetchStringTokenData(tokenContract, provider, 'symbol'),
+      tokenContract.decimals(),
+    ])
+
     return {
       address: getAddress(await tokenContract.getAddress()),
       name,
       symbol,
-      decimals,
+      decimals: Number(decimal),
     }
   } catch (error) {
     const errorMessage = extractErrorMessage(error)
     logger.warn(
       { tokenAddress, chainId, errorMessage },
       'Failed to fetch token metadata on-chain',
     )
-    return undefined
+    return {
+      address: getAddress(await tokenContract.getAddress()),
+      name: DEFAULT_STRING,
+      symbol: DEFAULT_STRING,
+      decimals: DEFAULT_DECIMAL,
+    }
```

Comment:
Problem here. On line 51 of this same file we check if the result from here is undefined and throw an error. That cannot happen any longer after this change.

If we are going to change that behaviour we have to:
- Remove the code that checks if this result is undefined, as it's no longer the case
- Make sure that logic is not needed elsewhere.

My main worry about doing this, is that if there's a hiccup when we are generating metadata, this will just simply create tokens with empty addresses.